c7n/resources/appflow.py

```diff
@@ -32,6 +35,77 @@ class resource_type(TypeInfo):
     source_mapping = {'describe': AppFlowDescribe, 'config': ConfigSource}
 
 
+@AppFlow.filter_registry.register('kms-key')
+class AppFlowKmsKeyFilter(ValueFilter):
```

Comment:
We _should_ be able to derive from [KmsRelatedFilter](https://github.com/cloud-custodian/cloud-custodian/blob/3b7e528c17f1e28799d19591143dc902e313e7a5/c7n/filters/kms.py#L8) here instead, and then the body of the class becomes:

```python
    RelatedIdsExpression = 'kmsArn'
```

Tried making that change locally before suggesting it, but it does look for aliases in the recording data that way and I don't have local app flow resources handy to re-record. Worth a shot though.